src/App.tsx

```diff
@@ -42,16 +45,52 @@ const App: React.FC = (): ReactElement => {
 
   function parseAttemptHistoryFromFile(fileObject: any): Attempt[] {
     const history: Attempt[] = fileObject?.Run?.AttemptHistory?.[0]?.Attempt?.flatMap(
-      ({ $, RealTime }: any) => {
-        return $?.id && RealTime?.[0]
-          ? { x: Number($.id), y: moment.duration(RealTime[0]).as('milliseconds') }
+      ({ id, RealTime }: any) => {
+        return id && RealTime
+          ? { x: Number(id), y: moment.duration(RealTime).as('milliseconds') }
           : []
       }
     )
 
     return history
   }
 
+  function parseSegmentsFromFile(fileObject: any): Segment[] {
+    const segmentData: Segment[] = fileObject?.Run?.Segments?.[0]?.Segment?.flatMap(
+      ({ Name, SplitTimes, BestSegmentTime, SegmentHistory }: any) => {
+        let name: string | null = null
+        if (Name?.[0]) {
+          name = Name[0]
+        }
+
+        let pbTime: number | null = null
+        if (
+          SplitTimes?.[0]?.SplitTime?.[0]?.name?.[0] === 'Personal Best' &&
+          SplitTimes?.[0]?.SplitTime?.[0]?.RealTime
+        ) {
+          pbTime = moment.duration(SplitTimes[0].SplitTime[0].RealTime).as('milliseconds')
+        }
+
+        let goldSplit: number | null = null
+        if (BestSegmentTime?.[0]?.RealTime) {
+          goldSplit = moment.duration(BestSegmentTime[0].RealTime).as('milliseconds')
+        }
+
+        let history: Attempt[] | null = SegmentHistory?.[0]?.Time?.flatMap(
+          ({ id, RealTime }: any) => {
+            return id && RealTime
+              ? { x: Number(id), y: moment.duration(RealTime).as('milliseconds') }
+              : []
+          }
+        )
+        history = history === undefined ? null : history
+
+        return { name, pbTime, goldSplit, history }
```

Comment:
If you're returning an entry for every item in the Segment array, then this can likely be a standard `map` instead of a `flatMap` - which provides the filtering/adding of items to the output array.

Reply:
Updated